### PR TITLE
chore(cypress): add e2e CI testing with cypress

### PR DESCRIPTION
**Pull request by ixio**
This is the base of working tests, there needs to be many more tests and this allows us to write them with Cypress.

### CODE REVIEW

File: frontend/cypress/e2e/0-osmose-tests/login.cy.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    cy.visit('http://localhost:3000/')
+  })
+
+  it('logs in with right credentials', () => {
+    cy.get("#loginInput").click();
+    cy.get("#loginInput").type("admin");
+    cy.get("#passwordInput").click();
+    cy.get("#passwordInput").type("osmose29");
+    cy.get("form > input").click();
+    cy.get("h1").should('have.text', 'APLOSEDatasets')
```
Comment (ElodieENSTA) [inline frontend/cypress/e2e/0-osmose-tests/login.cy.js:12]: I'm pretty sure there is an error in the text

File: frontend/cypress/e2e/0-osmose-tests/login.cy.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+describe('login page', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000/')
+  })
+
+  it('logs in with right credentials', () => {
```
Comment (ElodieENSTA) [inline frontend/cypress/e2e/0-osmose-tests/login.cy.js:6]: logs in **as admin**

File: frontend/cypress/examples/2-advanced-examples/actions.cy.js [hunk 3]
```diff
@@ -0,0 +1,299 @@
+/// <reference types="cypress" />
```
Comment (ElodieENSTA) [inline frontend/cypress/examples/2-advanced-examples/actions.cy.js:1]: same - shouldn't be in the repo

File: frontend/cypress/examples/2-advanced-examples/aliasing.cy.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/// <reference types="cypress" />
```
Comment (ElodieENSTA) [inline frontend/cypress/examples/2-advanced-examples/aliasing.cy.js:1]: same - shouldn't be in the repo

File: frontend/cypress/examples/2-advanced-examples/connectors.cy.js
Comment (ElodieENSTA) [inline frontend/cypress/examples/2-advanced-examples/connectors.cy.js:1]: same - shouldn't be in the repo

File: frontend/cypress/examples/2-advanced-examples/assertions.cy.js
Comment (ElodieENSTA) [inline frontend/cypress/examples/2-advanced-examples/assertions.cy.js:1]: same - shouldn't be in the repo

File: website/package-lock.json
Comment (ElodieENSTA) [inline website/package-lock.json:1]: Why is there updates on the website package-lock.json  but none on its package.json ?

File: frontend/cypress/e2e/0-osmose-tests/login.cy.js (hunk 1 above)
Comment (ixio) [reply frontend/cypress/e2e/0-osmose-tests/login.cy.js:12]: On that page we have two H1 titles, which is not a great idea and we should fix. There is a main H1 APLOSE title and then a second page-specific H1 Datasets title. When using `get=>have.text` you get the text of all the hits, so here `APLOSEDatasets`. We could try to get the second h1 only but a better long term change is removing the H1 title or moving page-titles to H2.

File: website/package-lock.json
Comment (ixio) [reply website/package-lock.json:1]: It's not updates, it's adding the whole file. See the .gitignore modification in the [We've long passed the production stage ](https://github.com/Project-OSmOSE/osmose-app/pull/166/commits/3fc751e95c9e8ba8bf6ad13a2796be3a388dcd81) commit

**Review by ElodieENSTA — APPROVED** [review]

**Pull request merged**